spec/pg_eventstore/queries/subscription_command_queries_spec.rb

```diff
@@ -0,0 +1,116 @@
+
+    let(:command_name) { 'DoSomething' }
+
+    context 'when command exists' do
+      let!(:command) { instance.create_by(subscription_id: subscription.id, command_name: command_name) }
+
+      it 'raises error' do
+        expect { subject }.to raise_error(PG::UniqueViolation)
+      end
+    end
+
+    context 'when command does not exist' do
+      it 'creates it' do
+        expect { subject }.to change {
+          instance.find_by(subscription_id: subscription.id, command_name: command_name)
+        }.to(a_hash_including(:id, :name, :subscription_id, :created_at))
+      end
+      it 'has correct attributes' do
+        aggregate_failures do
+          expect(subject[:id]).to be_a(Integer)
+          expect(subject[:name]).to eq(command_name)
+          expect(subject[:subscription_id]).to eq(subscription.id)
+          expect(subject[:created_at]).to be_between(Time.now.utc - 1, Time.now.utc + 1)
+        end
+      end
+    end
+
+    context 'when command with the same name exist, but for different Subscription' do
+      let(:another_subscription) { SubscriptionsHelper.create(set: 'BarSet') }
+      let!(:command) { instance.create_by(subscription_id: another_subscription.id, command_name: command_name) }
+
+      it 'creates new command' do
+        expect { subject }.to change {
+          instance.find_by(subscription_id: subscription.id, command_name: command_name)
+        }.to(a_hash_including(:id, :name, :subscription_id, :created_at))
+      end
+    end
+  end
+
+  describe '#find_commands' do
+    subject { instance.find_commands(subscription_ids) }
+
+    let(:subscription_ids) { [] }
+
+    context 'when subscription ids is empty' do
+      it { is_expected.to eq([]) }
+    end
+
+    context 'when subscription ids are present' do
+      let(:subscription_ids) { [subscription.id, subscription3.id] }
+
+      let(:subscription2) { SubscriptionsHelper.create(set: 'Bar') }
+      let(:subscription3) { SubscriptionsHelper.create(set: 'Baz') }
+      let!(:command1) { instance.create_by(subscription_id: subscription.id, command_name: 'Foo') }
+      let!(:command2) { instance.create_by(subscription_id: subscription2.id, command_name: 'Foo') }
+      let!(:command3) { instance.create_by(subscription_id: subscription3.id, command_name: 'Bar') }
+
+      it 'returns existing commands by the given subscription ids' do
+        is_expected.to eq([command1, command3])
+      end
+    end
+  end
+
+  describe '#delete' do
+    subject { instance.delete(id) }
+
+    let(:id) { -1 }
+
+    context 'when command exists' do
+      let(:id) { command[:id] }
+      let!(:command) { instance.create_by(subscription_id: subscription.id, command_name: 'Foo') }
+
+      it 'deletes it' do
+        expect { subject }.to change { instance.find_by(subscription_id: subscription.id, command_name: 'Foo') }.to(nil)
+      end
+    end
+
+    context 'when command does not exist' do
+      it 'does not thing' do
```

Comment:
I would put  `let(:id) { -1 }` here you don't need it in the line 99

Reply:
I need it there because it is used in `subject`. If someone would ever want to add another test - it won't fail with `NoMethodError` at least because of the missing `let(:id) {...}`

Reply:
Yes but for the set of spec currently you have this is not necessary to have it in that place